File: backend/src/HowAreMyFinances.Api/Models/Import.cs
namespace HowAreMyFinances.Api.Models;

public sealed record ImportRequest(
    IReadOnlyList<string> Categories,
    IReadOnlyList<ImportMonthEntry> Months
);

public sealed record ImportMonthEntry(
    int Year,
    int Month,
    decimal Salary,
    IReadOnlyList<ImportExpenseEntry> Expenses,
    IReadOnlyList<ImportIncomeEntry> Incomes
);

public sealed record ImportExpenseEntry(
    string ItemName,
    decimal Amount,
    string CategoryName,
    string? Vendor,
    string ExpenseDate,
    string? Comment
);

public sealed record ImportIncomeEntry(
    string Source,
    decimal Amount,
    string IncomeDate,
    string? Comment
);

public sealed record ImportResult(
    int CategoriesCreated,
    int MonthsCreated,
    int ExpensesCreated,
    int IncomesCreated
);

File: shared/types/import.ts
export interface ImportExpense {
  itemName: string;
  amount: number;
  categoryName: string;
  vendor?: string;
  expenseDate: string;
  comment?: string;
}

export interface ImportIncome {
  source: string;
  amount: number;
  incomeDate: string;
  comment?: string;
}

export interface ImportMonth {
  year: number;
  month: number;
  salary: number;
  expenses: ImportExpense[];
  incomes: ImportIncome[];
}

export interface ImportRequest {
  categories: string[];
  months: ImportMonth[];
}

export interface ImportResult {
  categoriesCreated: number;
  monthsCreated: number;
  expensesCreated: number;
  incomesCreated: number;
}
